test/controllers/votes_controller_test.rb

```diff
@@ -0,0 +1,71 @@
+require "test_helper"
+
+describe VotesController do
+  describe "create" do
+    it "is possible to vote if logged in" do
+      user = perform_login(User.first)
```

Comment:
Good work on the tests for this - the logic here is rather complex.

Organizationally, these target the `upvote_work_path`, which is under the `WorksController`, so I would probably put these tests in that test file.